Adapter/PhpcrOdmAdapter.php

```diff
@@ -190,12 +203,42 @@ public function getReferringAutoRoutes($contentDocument)
     public function findRouteForUri($uri, UriContext $uriContext)
     {
         $path = $this->getPathFromUri($uri);
-
-        return $this->dm->find(null, $path);
+        $node = $this->dm->find(null, $path);
+        if ($node instanceof AutoRouteInterface) {
+            return $node;
+        }
+        return false;
     }
 
     private function getPathFromUri($uri)
     {
         return $this->baseRoutePath . $uri;
     }
+
+    /**
+     * @param Generic $node
+     * @param object $contentDocument
+     * @param string $autoRouteTag
+     * @param string $routeType
+     * @return AutoRouteInterface
+     */
+    private function convertGenericNodeInAutoRouteNode(Generic $node, $contentDocument, $autoRouteTag, $routeType)
+    {
+        $autoRouteClassName = $this->autoRouteFqcn;
+        $mapper = $this->dm->getConfiguration()->getDocumentClassMapper();
+        $mapper->writeMetadata($this->dm, $node->getNode(), $autoRouteClassName);
```

Comment:
@dbu we have discussed migrating document types before -- is this a solution? I seem to remember that it was not so easy.